library/MachineControl.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from urllib.parse import quote
+
+from library.common import *
+from library.NetworkControl import NetworkControl
+from library.StateControl import StateControl
+
+class MachineControl(object):
+    def __init__(self, requester, machine_name):
+        self.requester = requester
+
+        try:
+            self.machine_id = self._get_id(machine_name)
+            self.networkcontrol = NetworkControl(self.requester, self.machine_id)
+        except Exception as err:
+            raise err
+
+        self.statecontrol = StateControl(self.requester, self.machine_id)
+
+    def _get_id(self, machine_name):
+        """ Look up machine by name """
+        q = '(= name "{}")'.format(quote(machine_name))
+        url = "/api/v1/machine?q={}&show_all=false".format(q)
+        try:
+            data = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        return data[0]['id']
```

Comment:
This will crash if nothing was returned, but the query "succeeded" in returning an empty result. You need to check for type and values, raising TypeError/ValueError.

Also, this would be a lot more useful if you would return the whole dictionary and let the caller select what it wants from it.